src/index.js

```diff
@@ -1,4 +1,168 @@
 'use strict'
 
-exports.IPLDService = require('./ipld-service')
-exports.resolve = require('./resolve')
+const Block = require('ipfs-block')
+const pull = require('pull-stream')
+const traverse = require('pull-traverse')
+const utils = require('./utils')
+const CID = require('cids')
+const until = require('async/until')
+const IPFSRepo = require('ipfs-repo')
+const MemoryStore = require('../node_modules/interface-pull-blob-store/lib/reference.js')
+const BlockService = require('ipfs-block-service')
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+
+class IPLDResolver {
+  constructor (blockService) {
+    // nicola will love this!
+    if (!blockService) {
+      const repo = new IPFSRepo('in-memory', { stores: MemoryStore })
+      blockService = new BlockService(repo)
+    }
+
+    this.bs = blockService
+    this.resolvers = {}
+
+    // Support by default dag-pb and dag-cbor
+    this.support(dagPB.resolver.multicodec, dagPB.resolver, dagPB.util)
+    this.support(dagCBOR.resolver.multicodec, dagCBOR.resolver, dagCBOR.util)
```

Comment:
What about having functions to add and remove, so that:

- `this.support.add`
- `this.support.rm`